entities/inverted/errors.go

```diff
@@ -0,0 +1,40 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package inverted
+
+import "fmt"
+
+type MissingIndexError struct {
+	message string
+}
+
+func NewMissingFilterableIndexError(propName string) error {
+	return MissingIndexError{message: fmt.Sprintf("Filtering by property '%s' requires inverted index. "+
```

Comment:
Nice to have:
It'd be nice if we could factor out the error message templates out of the functions so that no new format string is allocated when calling the New***.